packages/xdl/src/project/publishAsync.ts

```diff
@@ -0,0 +1,459 @@
+import {
+  ExpoAppManifest,
+  ExpoConfig,
+  getDefaultTarget,
+  PackageJSONConfig,
+  Platform,
+} from '@expo/config';
+import { bundleAsync } from '@expo/dev-server';
+import axios from 'axios';
+import chalk from 'chalk';
+import FormData from 'form-data';
+import fs from 'fs-extra';
+import path from 'path';
+
+import Analytics from '../Analytics';
+import ApiV2 from '../ApiV2';
+import Config from '../Config';
+import * as EmbeddedAssets from '../EmbeddedAssets';
+import { ErrorCode } from '../ErrorCode';
+import logger from '../Logger';
+import { publishAssetsAsync } from '../ProjectAssets';
+import * as Sentry from '../Sentry';
+import * as UrlUtils from '../UrlUtils';
+import UserManager, { User } from '../User';
+import XDLError from '../XDLError';
+import * as ExponentTools from '../detach/ExponentTools';
+import * as TableText from '../logs/TableText';
+import { learnMore } from '../logs/TerminalLink';
+import { resolveEntryPoint } from '../tools/resolveEntryPoint';
+import * as Doctor from './Doctor';
+import * as ProjectUtils from './ProjectUtils';
+import { getPublishExpConfigAsync, PublishOptions } from './getPublishExpConfigAsync';
+import { LoadedHook, prepareHooks, runHook } from './runHook';
+import { shouldUseDevServer } from './startDevServerAsync';
+import {
+  startReactNativeServerAsync,
+  stopReactNativeServerAsync,
+} from './startLegacyReactNativeServerAsync';
+
+const MINIMUM_BUNDLE_SIZE = 500;
+
+type PackagerOptions = {
+  dev: boolean;
+  minify: boolean;
+};
+
+export async function buildPublishBundlesAsync(
```

Comment:
This is used by both export and publish, so I think it would be better to put it in a separate file (`buildPublishBundlesAsync.ts`).